### PR TITLE
Add prompt constraints block to meal planning agent (SPEC-007 W6)

### LINKED ISSUE #338: SPEC-007 W6: meal-planner prompt constraints block
Parent: #212 · Plan: `system_design/plans/PLAN-007-nutrition-guardrail-pipeline.md` · Spec: SPEC-007 §4.6

## Scope

Add a structured constraints block to the `MealPlanningAgent` system prompt derived from `profile.restriction_resolution`. Prompt-side prevention complements the guardrail (enforcement); explicitly tells the model "if unsure, do not include — we reject anyway."

## Files

- `backend/agents/nutrition_meal_planning_team/agents/meal_planning_agent/agent.py` — extend `SYSTEM_PROMPT` and `_build_user_prompt`.
- `backend/agents/nutrition_meal_planning_team/agents/meal_planning_agent/prompt_constraints.py` — helper to render `RestrictionResolution` → the FORBIDDEN / FLAG-ONLY / SHORTHANDS block.
- `backend/agents/nutrition_meal_planning_team/agents/meal_planning_agent/tests/test_prompt_constraints.py`.

## Acceptance

- [ ] Constraints block renders deterministically from a `RestrictionResolution`.
- [ ] Empty resolution → no block (no prompt bloat).
- [ ] Snapshot tests cover: vegan, tree-nut allergy, warfarin patient, pescatarian-except-fish.

## Dependencies

None (independent of W1–W5).

## Priority

P0.

Refs #212.

### PR DESCRIPTION
**Pull request by brandonkindred**
## Summary

- Add `prompt_constraints.py` — pure function that renders a structured FORBIDDEN/FLAG-ONLY/EXEMPTIONS/SHORTHANDS constraints block from `RestrictionResolution` + `ClinicalInfo.medications`
- Modify `agent.py` — inject constraints block into the user prompt (between profile JSON and nutrition plan), add system-prompt instruction to obey FORBIDDEN entries
- Extract inline prompt construction into `_build_user_prompt()` for testability
- Add 31 tests covering: empty resolution, vegan, tree-nut allergy, warfarin, MAOI, pescatarian (with exemptions), pescatarian+fish-allergy paradox, combined profiles, soft constraints, ambiguous/unresolved restrictions, freetext medications, determinism (100 iterations), and overlapping medication flag deduplication

Defense-in-depth: the prompt tells the LLM what to avoid so it produces fewer violations that the guardrail must reject post-generation. Empty profiles get no constraints block (no prompt bloat).

## Test plan

- [x] 31 new tests in `test_prompt_constraints.py` — all pass
- [x] 83 existing guardrail tests — zero regressions
- [x] `ruff check` + `ruff format --check` clean
- [x] Determinism verified: 100 iterations byte-identical for complex profile

Closes #338

https://claude.ai/code/session_01Y4guJWf7mPpztQaHC3Emtu

---
_Generated by [Claude Code](https://claude.ai/code/session_01Y4guJWf7mPpztQaHC3Emtu)_

### CODE REVIEW

File: backend/agents/nutrition_meal_planning_team/agents/meal_planning_agent/agent.py [hunk 1]
```diff
@@ -73,19 +105,8 @@ def run(
...
-            + history_summary
-            + "\n\nRequest: suggest meals for the next "
-            + str(period_days)
-            + " days, meal types: "
-            + ", ".join(meal_types)
-            + '. Output JSON: {"suggestions": [ ... ]} with each item having name, ingredients, portions_servings, prep_time_minutes, cook_time_minutes, rationale, meal_type, suggested_date.'
-            + "\n\nRespond with valid JSON only, no markdown fences."
+        prompt = _build_user_prompt(
+            profile, nutrition_plan, history_summary, period_days, meal_types
         )
```
[the remark this answers is not in the document]
Comment (brandonkindred) [reply backend/agents/nutrition_meal_planning_team/agents/meal_planning_agent/agent.py:113]: Good catch. Moved `_build_user_prompt()` inside the existing `try/except` so that `InteractionDataError` from `get_interaction_policies()` is caught and the agent returns `[]` instead of propagating — consistent with the pre-existing failure behavior. Fixed in 759871e.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by brandonkindred** [comment]
@codex review

File: backend/agents/nutrition_meal_planning_team/agents/meal_planning_agent/agent.py [hunk 2]
```diff
@@ -73,22 +105,10 @@ def run(
...
-            + " days, meal types: "
-            + ", ".join(meal_types)
-            + '. Output JSON: {"suggestions": [ ... ]} with each item having name, ingredients, portions_servings, prep_time_minutes, cook_time_minutes, rationale, meal_type, suggested_date.'
-            + "\n\nRespond with valid JSON only, no markdown fences."
-        )
-
         try:
+            prompt = _build_user_prompt(
+                profile, nutrition_plan, history_summary, period_days, meal_types
+            )
```
[the remark this answers is not in the document]
Comment (brandonkindred) [reply backend/agents/nutrition_meal_planning_team/agents/meal_planning_agent/agent.py]: Good point — split into two separate `try/except` blocks. Prompt-build failures now log at `logger.error("Prompt constraint build failed: ...")` while LLM call failures remain at `logger.warning("Meal planning LLM call failed: ...")`. Both still return `[]` to the caller, but the log level and message now distinguish configuration/data-integrity problems from transient LLM failures. Fixed in 99c670f.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by brandonkindred** [comment]
@codex review

File: backend/agents/nutrition_meal_planning_team/agents/meal_planning_agent/agent.py [hunk 3]
```diff
@@ -73,20 +105,13 @@ def run(
...
-            + '. Output JSON: {"suggestions": [ ... ]} with each item having name, ingredients, portions_servings, prep_time_minutes, cook_time_minutes, rationale, meal_type, suggested_date.'
-            + "\n\nRespond with valid JSON only, no markdown fences."
-        )
+        try:
+            prompt = _build_user_prompt(
+                profile, nutrition_plan, history_summary, period_days, meal_types
+            )
+        except Exception as e:
+            logger.error("Prompt constraint build failed: %s", e)
+            return []
```
[the remark this answers is not in the document]
Comment (brandonkindred) [reply backend/agents/nutrition_meal_planning_team/agents/meal_planning_agent/agent.py]: Agreed — constraints are defense-in-depth, not a gate. Moved the `try/except` into `_build_user_prompt` around the `render_constraints_block` call itself. If rendering fails, it logs at `ERROR` and continues building the prompt without the constraints block. The LLM proceeds normally and the downstream guardrail checker still catches violations. Fixed in 2c290ba.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by brandonkindred** [comment]
@codex review

File: backend/agents/nutrition_meal_planning_team/agents/meal_planning_agent/agent.py [hunk 4]
```diff
@@ -55,6 +56,40 @@ def _summarize_history(entries: List[MealHistoryEntry]) -> str:
...
+) -> str:
+    """Assemble the user prompt with optional constraints block."""
+    parts = ["Client profile:\n" + profile.model_dump_json(indent=2)]
+
+    try:
+        constraints = render_constraints_block(profile.restriction_resolution, profile.clinical)
+        if constraints:
+            parts.append(constraints)
+    except Exception as e:
+        logger.error("Prompt constraint rendering failed, continuing without constraints: %s", e)
```
[the remark this answers is not in the document]
Comment (brandonkindred) [reply backend/agents/nutrition_meal_planning_team/agents/meal_planning_agent/agent.py:74]: This contradicts the previous review comment which asked for graceful degradation instead of failing closed. The current approach is the right tradeoff:

1. **Allergen and dietary constraints** — fully enforced by `guardrail/checker.py` downstream regardless of whether the prompt block renders. The checker is the enforcement layer; the prompt is defense-in-depth.

2. **Medication interactions** — not yet enforced in the checker (that's W4 / #336), but `interactions.yaml` ships with the codebase and is `@lru_cache`-loaded. If it's missing, deployment is fundamentally broken and will surface immediately on any call. This isn't a transient failure that silently degrades over time.

3. **Impact of failing closed** — returning `[]` for any user with medications listed (which is common) would be strictly worse than proceeding without the prompt hint. In v1, only MAOI × tyramine is a hard interaction; everything else is advisory.

The `logger.error` ensures the configuration failure is visible in monitoring. Keeping the graceful degradation.

---
_Generated by [Claude Code](https://claude.ai/code)_
